compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -2444,6 +2446,63 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
         }
     }
 
+    fn undeclared_module_suggest_declare(
+        &mut self,
+        ident: Ident,
+        path: std::path::PathBuf,
+    ) -> Option<(Vec<(Span, String)>, String, Applicability)> {
+        let map = self.tcx.sess.source_map();
+        let sp = map
+            .span_extend_to_prev_char(
+                map.span_extend_to_prev_char(ident.span, '\n', true),
+                '\n',
+                true,
+            )
+            .shrink_to_lo();
+        Some((
+            vec![(sp, format!("mod {ident};\n"))],
+            format!(
+                "to make use of source file {}, use `mod {ident}` \
+in this file to declare the module",
```

Comment:
```suggestion
                 in this file to declare the module",
```
The whitespace after `\` is ignored.